src/cli.rs

```diff
@@ -0,0 +1,501 @@
+                                    None => {
+                                        eprintln!("A period value should be available");
+                                    }
+                                    Some(p) => match sig.verify(p, &pk, msg) {
+                                        Ok(()) => {
+                                            println!("OK");
+                                        }
+                                        _ => {
+                                            println!("Fail");
+                                        }
+                                    },
+                                }
+                            }
+                            Err(err) => {
+                                eprintln!("Decode error of the secret key: {err}");
+                            }
+                        }
+                    }
+                },
+            };
+        }
+        Cmd::UpdateSk => {
+            match config.file {
+                None => {
+                    eprintln!("No stdin or file was provided to read a signing key");
+                }
+                Some(sk_source) => match open_any(&sk_source) {
+                    Err(err) => {
+                        eprintln!("Failed to open {sk_source}: {err}");
+                    }
+                    Ok(sk_handle) => {
+                        let mut buffer = [0; 1224];
+                        let mut handle = sk_handle.take(1224);
+                        handle.read_exact(&mut buffer)?;
+                        match hex::decode(buffer) {
+                            Ok(bs) => {
+                                let mut sk_bytes = [0u8; 612];
+                                sk_bytes.copy_from_slice(&bs);
+                                match Sum6Kes::from_bytes(&mut sk_bytes) {
+                                    Ok(mut sk) => match sk.update() {
+                                        Ok(()) => {
+                                            print!("{}", hex::encode(sk.as_bytes()));
+                                        }
+                                        _ => {
+                                            eprintln!("Signing key failed to be updated");
+                                        }
+                                    },
+                                    _ => {
+                                        eprintln!("Signing key expects 612 bytes");
+                                    }
+                                };
+                            }
+                            Err(err) => {
+                                eprintln!("Decode error of the signing key: {err}");
+                            }
+                        }
+                    }
+                },
+            };
+        }
+    }
+    Ok(())
+}
+
+/// Parses line entered by user into config
+pub fn get_args() -> CLIResult<Config> {
+    let matches = App::new("kes-summed-ed25519")
+        .version("0.1.0")
```

Comment:
Probably want to automatically extract version and author from [Cargo.toml](https://github.com/pragma-org/amaru/blob/main/crates/amaru/src/bin/amaru/main.rs#L58).